packages/plugin-core/src/survey.ts

```diff
@@ -0,0 +1,173 @@
+  };
+
+  constructor(opts: {
+    choices: vscode.QuickPickItem[];
+    canPickMany: boolean;
+    title: string;
+    placeHolder?: string;
+  }) {
+    const { choices, canPickMany, title } = opts;
+    let placeHolder = opts.placeHolder;
+    this.choices = choices;
+
+    if (!placeHolder) {
+      placeHolder = canPickMany ? "Check all that applies." : "Check one";
+    }
+
+    this.opts = { title, placeHolder, canPickMany, ignoreFocusOut: true };
+  }
+
+  getChoices(): readonly vscode.QuickPickItem[] {
+    return this.choices;
+  }
+
+  onAnswer(_opts: any): void {
+    return undefined;
+  }
+  onReject(_opts?: any): void {
+    return undefined;
+  }
+
+  async show() {
+    const results = await vscode.window.showQuickPick(this.choices, this.opts);
+    if (results) {
+      this.onAnswer(results);
+    } else {
+      this.onReject();
+    }
+
+    return results;
+  }
+}
+
+export class BackgroundSurvey extends DendronSurvey {
+  onAnswer(results: vscode.QuickPickItem[]) {
+    AnalyticsUtils.track(SurveyEvents.BackgroundAnswered, {
+      results: results.map((result) => result.label),
+    });
+  }
+
+  onReject() {
+    AnalyticsUtils.track(SurveyEvents.BackgroundRejected);
+  }
+
+  static create() {
+    const title = "What is your background?";
+    const choices = [
+      { label: "Software Developer" },
+      { label: "Technical Writer" },
+      { label: "Dev Ops" },
+      { label: "Manager" },
+      { label: "Student" },
+      { label: "Other" },
+    ];
+    return new BackgroundSurvey({ title, choices, canPickMany: true });
+  }
+}
+
+export class UseCaseSurvey extends DendronSurvey {
+  onAnswer(results: vscode.QuickPickItem[]) {
+    AnalyticsUtils.track(SurveyEvents.UseCaseAnswered, {
+      results: results.map((result) => result.label),
+    });
+  }
+
+  onReject() {
+    AnalyticsUtils.track(SurveyEvents.UseCaseRejected);
+  }
+
+  static create() {
+    const title = "Tell us what you want to use Dendron for";
+    const choices = [
+      { label: "Team knowledge base" },
+      { label: "Todos and Agenda" },
+      { label: "Meeting notes" },
+      { label: "Publishing" },
+      { label: "Research" },
+      { label: "Other" },
+    ];
+    return new UseCaseSurvey({ title, choices, canPickMany: true });
+  }
+}
+
+export class PriorToolsSurvey extends DendronSurvey {
+  onAnswer(results: vscode.QuickPickItem[]) {
+    AnalyticsUtils.track(SurveyEvents.PriorToolsAnswered, {
+      results: results.map((result) => result.label),
+    });
+  }
+
+  onReject() {
+    AnalyticsUtils.track(SurveyEvents.PriorToolsRejected);
+  }
+
+  static create() {
+    const title = "Are you coming from an existing tool(s)?";
+    const choices = [
+      { label: "No" },
+      { label: "Foam" },
+      { label: "Roam" },
+      { label: "Logseq" },
+      { label: "Notion" },
+      { label: "OneNote" },
+      { label: "Obsidian" },
+      { label: "Evernote" },
+      { label: "Google Keep" },
```

Comment:
Add Other option here too?

Reply:
yeah that's a bit tricky to do with the current quickpick. especially if we are allowing `canSelectMany`. If we want to have "other", then we should have another quick input chained after this

Reply:
I've added a check in `onAnswer` to see if the user checked `Other`, and made it prompt a quick input to manually type in an answer. Looks a bit hacky but I think this is good enough for now.